---
title: "Evolutionary analyses of mobster fits: dnds and ..."
author: "Giulio Caravagna"
date: "September 2019"
institute: "Institute for Cancer Research"
email: "giulio.caravagna@icr.ac.uk"
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

This vignette describes the evolutionary analyses that are available in [MOBSTER](https://github.com/caravagn/MOBSTER), and include:

* computation of dnds values via the package `dndscv`;
* computation of evolutionary parameters;

```{r, message=FALSE, warning=F}
library(mobster)
```

# Computing dnds values

Dn/Ds values are computed wrapping a call to the [dndscv](https://github.com/im3sanger/dndscv/) R package, developed by Martincorena at Sanger.

> Martincorena, et al. "Universal patterns of selection in cancer and somatic tissues." Cell 171.5 (2017): 1029-1041.

These values can only be computed if the available mutation calls  have the following information:

* the chromosome location, labelled as `chrom`;
* the mutation position, labelled as `from`;
* the mutant and the reference alleles, labelled as `alt` and `ref`.

To show these analysis we use the fits for the `PD4120a` breast cancer sample available in  MOBSTER.

```{r, fig.width=5, fig.height=4}
data('PD4120a_breast_sample', package = 'mobster')

# Print and plot the model
print(PD4120a_breast_sample$best)
plot(PD4120a_breast_sample$best)
```

To compute the values one first extracts the clustering assignments from the best fit.

```{r, fig.width=2.5, fig.height=5}
clusters = Clusters(PD4120a_breast_sample$best)
pio::pioDisp(clusters)
```

The available clusters are `C1`, `C2`, `C3` and `Tail`; `C1` is the clonal cluster and 
`C2` and `C3` are subclones. 

To compute the dnds statistics one has to define a custom grouping on top of the actual clusters,
so that the statistics are computed by group.

Groups are defined from a `mapping` variable.

> If `mapping  = c(`C1` = 'G1', `C2` = 'G1', `C3` = 'G1', `Tail` = 'G2')`, 
then mutations from clusters `C1`, `C2` and `C3` will be pooled into one group (`G1`), 
while mutations from cluster `Tail` will constitute a group themselves. 
By default, with `mapping = NULL`, each cluster is a group. 

One can now run the analysis using this grouping and the default gene list (`gene_list = NULL`)
available in `dndscv`.

```{r, fig.width=3, fig.height=3, warning=FALSE}
# Run with this grouping and default gene list
dnds_stats = dnds(
  clusters,
  mapping = c(`C1` = 'Non-tail', `C2` = 'Non-tail', `C3` = 'Non-tail', `Tail` = 'Tail'),
  gene_list = NULL
)
```

The returned outputs are the tables returned by `dndscv`, as well as a `ggplot` plot for the object. The column `dnds_group` labels the group used to compute the statistics.
```{r, fig.width=3, fig.height=3, warning=FALSE}
# Summary statistics
print(dnds_stats$dnds_summary)

# Table observation countns
print(dnds_stats$dndscv_table)

# Plot
print(dnds_stats$plot)
```

By default, the plot contains all the results from the substitution models availavle in `dndscv`. If one wants to
visualize only the global dnds score parameter `dndscv_plot` should be set to `wall` in the call to `dnds()`.

## Using custom genes lists

A custom list of genes can be supplied in the call to `dnds()` as the variable `genes_list`.

MOBSTER provides 4 lists of interests for this type of computation:

* a list of driver genes compiled in `Martincorena et al. Cell 171.5 (2017): 1029-1041.`;
* a list of driver genes compiled in `Tarabichi, et al. Nature Genetics 50.12 (2018): 1630.`;
* a list of essential genes compiled in `Wang et al. Science 350.6264 (2015): 1096-1101.`;
* a list of essential genes compiled in `Bloomen et al. Science 350.6264 (2015): 1092-1096.`.

which are available to load.

```{r}
# Load the list
data('cancer_genes_dnds', package = 'mobster')

# Each sublist is a list 
print(lapply(cancer_genes_dnds, head))
```

As suggestion, the list of driver genes by Tarabichi et al seems to work better with TCGA data. The two
lists of essential genes have beenn instead compiled from two different cell lines.

To use a custom gene list run the following. Notice that if one uses data from a single patient it can often happen that there are not enough substitutions to compute the dnds values; in this case an error is returned which might be circumvented pooling data from multiple patients.
```{r, eval=FALSE}
# Not run here
dnds_stats = dnds(
  clusters,
  mapping = c(`C1` = 'Non-tail', `C2` = 'Non-tail', `C3` = 'Non-tail', `Tail` = 'Tail'),
  gene_list = cancer_genes_dnds$Martincorena_drivers
)
```

## Pooling data from multiple patients

The input format of the `dnds` function allows to pool data from several MOBSTER fits.

Here we load and pool together some of the real datasets available in the package, and use the same grouping as shown above.

```{r}
data('LU4_lung_sample', package = 'mobster')
data('LUFF76_lung_sample', package = 'mobster')
data('PD4120a_breast_sample', package = 'mobster')
```

We pool the data before the call to `dnds`, subsetting the columns to the ones required for this computation.

```{r, warning=F}
dnds_multi = dnds(
  rbind(
    Clusters(LU4_lung_sample$best) %>% select(chr, from, ref, alt, cluster) %>% mutate(sample = 'LU4'),
    Clusters(LUFF76_lung_sample$best) %>% select(chr, from, ref, alt, cluster) %>% mutate(sample = 'LUFF76'),
    Clusters(PD4120a_breast_sample$best) %>% select(chr, from, ref, alt, cluster) %>% mutate(sample = 'PD4120a')
  ),
  mapping = c(`C1` = 'Non-tail', `C2` = 'Non-tail', `C3` = 'Non-tail', `Tail` = 'Tail'),
)

print(dnds_multi$plot)
```

