---
title: "mobster"
author: "Giulio Caravagna"
date: "January 2020"
institute: "Institute for Cancer Research"
email: "giulio.caravagna@icr.ac.uk"
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{mobster}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```


`mobster` implements a `(k+1)`-dimensional Dirichlet mixture model to detect ongoing positive subclonal  selection from cancer genome sequencing data. The models performs a deconvolution of the site/ allele frequency spectrum of mutation data (signal).

The statistical model in `mobster` integrates principles from both *evolutionary theory* and *Machine Learning*. The mixture contains two types of distributions:

* Beta distributions (`k`), that capture `k` peaks of alleles raising up in frequency, as a proxy for subclone enjoying positive selection;
* One ParetoType-I power law for within-clone neutral dynamics, a  distribution predicted by theoretical Population Genetics.

This is a model-based method to analyse cancer data, where the "model" is the power law tail that is used to integrate evolutionary dynamics in this analysis. Results from `mobster` deconvolution can be used to reconstruct the clonal architecture of a tumour (subclonal deconvolution), focusing on potential patterns of functional heterogeneity. 

Available vignettes
-------

* [Introduction (data formats, example datasets)](https://caravagn.github.io/mobster/articles/introduction.html). 
* [Plotting mobster fits](https://caravagn.github.io/mobster/articles/plotting.html). 
* [dN/dS statistics of mobster fits](https://caravagn.github.io/mobster/articles/dnds.html). 
* [Population Genetics statistics of mobster fits](https://caravagn.github.io/mobster/articles/popgen.html). 
* [Confidence estimation via the bootstrap](https://caravagn.github.io/mobster/articles/bootstrap.html). 


