---
title: "mobster"
author: "Giulio Caravagna"
date: "September 2019"
institute: "Institute for Cancer Research"
email: "giulio.caravagna@icr.ac.uk"
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{mobster}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```


`mobster` implements a `(k+1)`-dimensional Dirichlet mixture model to detect statistical signals of ongoing subclonal evolution from cancer genome sequencing data. The models performs a deconvolution of the site/ allele frequency spectrum of the signal in order to detect non-neutral subclonal expansions.

The statistical model integrates principles from evolutionary theory and Machine-Learning. 

* Beta distributions (`k`) that capture `k` peaks originating possibly from alleles enjoying positive selection. 
* One power law theoretical distribution (the model) for within-clone neutral dynamics, predicted by Population Genetics;

The model is a mixture with `K>0` Beta and 1 Pareto  Type-I power law component, and can be used toreconstruct the clonal architecture of a tumour (subclonal deconvolution). Compared to alternative models for subclonal deconvolution, `mobster` is the first model to integrate neutral evolutionary dynamics in this analysis. 


Available vignettes
-------

* [Introduction (data formats, example datasets)](https://caravagn.github.io/mobster/articles/introduction.html). 
* [Plotting mobster fits](https://caravagn.github.io/CNAqc/articles/plotting.html). 
* [Evolutionary analysis of mobster fits](https://caravagn.github.io/CNAqc/articles/evolutionary_analysis.html). 
* [Confidence estimation via the bootstrap](https://caravagn.github.io/CNAqc/articles/bootstrap.html). 


