---
title: "mobster"
author: "Giulio Caravagna"
date: "January 2020"
institute: "Institute for Cancer Research"
email: "giulio.caravagna@icr.ac.uk"
output: rmarkdown::github_document
vignette: >
  %\VignetteIndexEntry{mobster}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

`mobster` implements a Dirichlet finite mixture model to detect ongoing positive subclonal selection from cancer genome sequencing data. The algorithm works best with high-resolution whole-genome sequencing data.

```{r, include = FALSE}
x = mobster::random_dataset(seed = 123, Beta_variance_scaling = 100, N = 2000)
x$data$VAF = (x$data$VAF / max(x$data$VAF)) / 1.6
x = mobster::mobster_fit(x$data, auto_setup = 'FAST')
```
```{r, fig.width=4, fig.height=3, echo=FALSE}
mobster::plot.dbpmm(x$best) + labs(caption = NULL)
```

The models performs a deconvolution of the site/ allele frequency spectrum of mutation data (the _signal_), and looks for models with `k+1` mixture components to fit the data. The plot above  shows a subclonal expansion (`C2`, at about `30%` allelic frequency) outgrowing an ancestral clonal population (`C1`, at `50%` allelic frequency for heterozygous mutations), as [expected from the interplay of positive and neutral selection forces](https://caravagn.github.io/mobster/articles/Example_tumour_simulation.html).

Inspired from both _mathematical modelling of evolutionary processes_ and _Machine Learning_, the  signal is modeled as mixture density with two types of distributions:

* `k` Betas to capture the peaks of alleles raising up in frequency in different clones (subclones enjoying positive selection, and the clonal cluster);
* `1` Pareto Type-I power law to model within-clone neutral dynamics, which is the distribution predicted by theoretical Population Genetics.

This is a _model-based_ approach to analyse cancer data, meaning that a power law tail is used to integrate evolutionary dynamics in this traditional clustering problem. Results from `mobster` deconvolution can be used to reconstruct the clonal architecture of a tumour (_subclonal deconvolution_) and identify patterns of functional heterogeneity (subclones under positive selection). 

Available vignettes
-------

We suggest you to [read about some example data](https://caravagn.github.io/mobster/articles/introduction.html) and then [move on to plot the fits](https://caravagn.github.io/mobster/articles/plotting.html) of your analysis. 

That is the basics, but if you want to do more you can use your fits to [compute dN/dS statistics](https://caravagn.github.io/mobster/articles/dnds.html) from the clusters, or [extract Population Genetics parameters](https://caravagn.github.io/mobster/articles/popgen.html) of tumour growth. You should also [assess the confidence of your estimates](https://caravagn.github.io/mobster/articles/bootstrap.html) via the bootstrap. 

For a real case study with `mobster`, check the [Supplementary Data ](https://caravagn.github.io/) released with our paper. 

