<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>1. Introduction • mobster</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="1. Introduction">
<meta property="og:description" content="">
<meta property="og:image" content="caravagn.github.io/mobster/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_plotting.html">2. Plotting fits</a>
    </li>
    <li>
      <a href="../articles/a3_bootstrap.html">3. Confidence estimation by the bootstrap</a>
    </li>
    <li>
      <a href="../articles/a4_popgen.html">4. Population Genetics statistics</a>
    </li>
    <li>
      <a href="../articles/a5_dnds.html">5. dN/dS statistics</a>
    </li>
    <li>
      <a href="../articles/a6_CloneTrees.html">6. Clone trees from clusters</a>
    </li>
    <li>
      <a href="../articles/a7_Example_tumour_simulation.html">7. Visualizing subclonal expansions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/mobster">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>1. Introduction</h1>
                        <h4 class="author">Giulio Caravagna</h4>
            
            <h4 class="date">03 May, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/mobster/blob/master/vignettes/a1_introduction.Rmd"><code>vignettes/a1_introduction.Rmd</code></a></small>
      <div class="hidden name"><code>a1_introduction.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(mobster)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(tidyr)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(dplyr)</a></code></pre></div>
<div id="input-data-for-mobster" class="section level1">
<h1 class="hasAnchor">
<a href="#input-data-for-mobster" class="anchor"></a>Input data for <code>mobster</code>
</h1>
<p>You can run a MOBSTER analysis if, for a set of input mutations (SNVs, indels etc.), you have available VAF or CCF data. The input data can be loaded using different input formats.</p>
<p>For VAF values you can use:</p>
<ul>
<li>a <code>data.frame</code> (or, <code>tibble</code>) with a column named <code>VAF</code> whose values are numerical <span class="math inline">\(0&lt;x&lt;1\)</span>, without <code>NA</code> entries;</li>
<li>a VCF file that must contain at least a column with the total depth of sequencing and the number of reads with the variants. In this case you need first to use function <code>load_VCF</code> and load the VCF content, and then proceed using your data as a <code>data.frame</code>.</li>
</ul>
<p>For CCF values you can only use the a <code>data.frame</code> format. Importantly, you have to store the CCF values in a column again named <code>VAF</code>, which must follow the same convention of a VAF column (i.e., range of values). Since CCF values usually peak at around <code>1.0</code> for clonal mutations (i.e., present in 100% of the input cells), we suggest to adjust standard CCF estimates dividing them by <code>0.5</code> in order to reflect the peak of an <em>heterozygous clonal mutation</em> at 50% VAF for a 100% pure bulk sample.</p>
<p><strong>Example dataset</strong>. Diploid mutations from sample <code>LU4</code> of the <a href="http://genome.kaist.ac.kr/">Comprehensive Omics Archive of Lung Adenocarcinoma</a> are available in the package under the name <code>LU4_lung_sample</code>. The available object is the results of an analysis with <code>mobster</code>, and the input mutation data is stored inside the object.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Example dataset LU4_lung_sample, downloaded from http://genome.kaist.ac.kr/</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(mobster<span class="op">::</span>LU4_lung_sample<span class="op">$</span>best<span class="op">$</span>data)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; # A tibble: 1,282 x 12</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt;      Key Callers t_alt_count t_ref_count Variant_Classif…    DP    VAF chr  </span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt;    &lt;int&gt; &lt;chr&gt;         &lt;int&gt;       &lt;int&gt; &lt;chr&gt;            &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt;  1   370 mutect…          30          95 intergenic         125 0.24   chr2 </span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt;  2   371 mutect…          10         105 intergenic         115 0.0870 chr2 </span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">#&gt;  3   372 mutect…          40         108 intronic           148 0.270  chr2 </span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">#&gt;  4   373 mutect…          20         101 intronic           121 0.165  chr2 </span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">#&gt;  5   374 mutect…          39          89 intergenic         128 0.305  chr2 </span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">#&gt;  6   375 mutect…          41         120 intronic           161 0.255  chr2 </span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">#&gt;  7   376 mutect…          43          93 intronic           136 0.316  chr2 </span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">#&gt;  8   377 mutect…          22          95 intergenic         117 0.188  chr2 </span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt;  9   378 mutect            7         104 intergenic         111 0.0631 chr2 </span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt; 10   379 mutect…          35          93 intergenic         128 0.273  chr2 </span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt; # … with 1,272 more rows, and 4 more variables: from &lt;chr&gt;, ref &lt;chr&gt;,</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt; #   alt &lt;chr&gt;, cluster &lt;chr&gt;</span></a></code></pre></div>
<p>Other datasets are available through the <code>data</code> command.</p>
<div id="driver-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#driver-annotations" class="anchor"></a>Driver annotations</h2>
<p>In the context of subclonal deconvolution we are often interested in linking “driver” events to clonal expansions. Since <code>mobster</code> works with somatic mutations data, it is possible to annotate the status of “driver mutation” in the input data; doing so, the drivers will be reported in some visualisations of the tool, but will not influence any of the computation carried out in <code>mobster</code>.</p>
<p>The annotate one or more <em>driver mutations</em> you need to include in your column 2 extra columns:</p>
<ul>
<li>
<code>is_driver</code>, a boolean <code>TRUE/FALSE</code> flag;</li>
<li>
<code>driver_label</code>, a character string that will be used as label in any visualisation that uses drivers.</li>
</ul>
</div>
</div>
<div id="generating-random-models-and-data" class="section level1">
<h1 class="hasAnchor">
<a href="#generating-random-models-and-data" class="anchor"></a>Generating random models and data</h1>
<p>You can sample a random dataset with the <code>random_dataset</code> function, setting:</p>
<ul>
<li>the number of mutations (<code>n</code>) and Beta components (<code>k</code>, subclones) to generate;</li>
<li>ranges and constraints on the size of the components;</li>
<li>ranges for the mean and variance of the Beta components.</li>
</ul>
<p>The variance of the Betas is defined as <span class="math inline">\(u/B\)</span> where <span class="math inline">\(u \sim U[0,1]\)</span>, and <span class="math inline">\(B\)</span> is the input parameter <code>Beta_variance_scaling</code>. Roughly, values of <code>Beta_variance_scaling</code> on the order of <code>1000</code> give low variance and sharp peaked data distributions. Values on the order of <code>100</code> give much wider distributions.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">dataset =<span class="st"> </span><span class="kw"><a href="../reference/random_dataset.html">random_dataset</a></span>(</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">  <span class="dt">seed =</span> <span class="dv">123456789</span>, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">  <span class="dt">Beta_variance_scaling =</span> <span class="dv">100</span>    <span class="co"># variance ~ U[0, 1]/Beta_variance_scaling</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">  )</a></code></pre></div>
<p>A list with 3 components is returned, which contains the actual data, sampled parameters of the generative model, and a plot of the data.</p>
<p>In <code>mobster</code> we provide the implementation of the model’s density function (<code>ddbpmm</code>, density Dirichlet Beta Pareto mixture model), and a sampler (<code>rdbpmm</code>) which is used internally by <code>random_dataset</code> to generate the data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Data, in the MOBSTER input format with a "VAF" column.</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dataset<span class="op">$</span>data)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; # A tibble: 5,000 x 2</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt;      VAF cluster</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt;    &lt;dbl&gt; &lt;chr&gt;  </span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt;  1 0.856 C1     </span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;  2 0.853 C1     </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt;  3 0.844 C1     </span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;  4 0.827 C1     </span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;  5 0.855 C1     </span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;  6 0.854 C1     </span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt;  7 0.878 C1     </span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt;  8 0.865 C1     </span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;  9 0.845 C1     </span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; 10 0.874 C1     </span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; # … with 4,990 more rows</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co"># The generated model contains the parameters of the Beta components (a and b),</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co"># the shape and scale of the tail, and the mixing proportion. </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dataset<span class="op">$</span>model)</a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; $a</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt;      C1      C2 </span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; 72.6286 30.5932 </span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt; $b</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt;       C1       C2 </span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt; 13.07363 24.93900 </span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt; $shape</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt; [1] 1</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt; $scale</span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt; [1] 0.05</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt; $pi</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt;      Tail        C1        C2 </span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt; 0.3431634 0.3331162 0.3237204</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co"># A plot object (ggplot) is available where each data-point is coloured by </span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="co"># its generative mixture component. The vertical lines annontate the means of</span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co"># the sampled Beta distributions.</span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(dataset<span class="op">$</span>plot)</a></code></pre></div>
<p><img src="a1_introduction_files/figure-html/unnamed-chunk-5-1.png" width="384"></p>
</div>
<div id="fitting-a-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#fitting-a-dataset" class="anchor"></a>Fitting a dataset</h1>
<p>Function <code>mobster_fit</code> fits a MOBSTER model.</p>
<p>The function implements a model-selection routine that by default scores models by their <code>reICL</code> (<em>reduced Integrative Classification Likelihood</em>) score, a variant to the popular <code>BIC</code> that uses the entropy of the latent variables of the mixture. <code>reICL</code> is discussed in the main paper.</p>
<p>This function has several parameters to customize the fitting procedure, and a set of special pre-parametrised runs that can be activated with parameter <code>auto_setup</code>. Here we use <code>auto_setup = "FAST"</code>, an automatic setup for a fast run; its parameters are accessible through an internal package function.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># Hidden function (:::)</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">mobster<span class="op">:::</span><span class="kw">template_parameters_fast_setup</span>()</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; $K</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; [1] 1 2</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; $samples</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; [1] 2</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; $init</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt; [1] "random"</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; $tail</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; [1]  TRUE FALSE</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; $epsilon</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; [1] 1e-06</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt; $maxIter</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; [1] 100</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; $fit.type</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; [1] "MM"</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt; $seed</span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt; [1] 12345</span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt; $model.selection</span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt; [1] "reICL"</span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt; $trace</span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt; [1] FALSE</span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; $parallel</span></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"><span class="co">#&gt; [1] FALSE</span></a>
<a class="sourceLine" id="cb5-35" data-line-number="35"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36"><span class="co">#&gt; $pi_cutoff</span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37"><span class="co">#&gt; [1] 0.02</span></a>
<a class="sourceLine" id="cb5-38" data-line-number="38"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-39" data-line-number="39"><span class="co">#&gt; $N_cutoff</span></a>
<a class="sourceLine" id="cb5-40" data-line-number="40"><span class="co">#&gt; [1] 10</span></a></code></pre></div>
<p>Compared to these, default parameters test more extensive types of fits (i.e., more clones, longer fits, higher number of replicates etc.). We usually use the fast parametrisation to obtain a first fit of the data and, if not satisfied, we run customised calls of <code>mobster_fit</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Fast run with auto_setup = "FAST"</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">fit =<span class="st"> </span><span class="kw"><a href="../reference/mobster_fit.html">mobster_fit</a></span>(</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">  dataset<span class="op">$</span>data,    </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">  <span class="dt">auto_setup =</span> <span class="st">"FAST"</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5">  )</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="co">#&gt;  [ MOBSTER fit ]</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co">#&gt; ✔ Loaded input data, n = 5000.</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="co">#&gt; ❯ n = 5000. Mixture with k = 1,2 Beta(s). Pareto tail: TRUE and FALSE. Output clusters with π &gt; 0.02</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co">#&gt; and n &gt; 10.</span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt; ! mobster automatic setup FAST for the analysis.</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; ❯ Scoring (without parallel) 2 x 2 x 2 = 8 models by reICL.</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; ℹ MOBSTER fits completed in 9.2s.</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 2 Beta(s) and a tail ─────────────────────────────────</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="co">#&gt; ● Clusters: π = 35% [C2], 33% [Tail] and 32% [C1], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="co">#&gt; ● Tail [n = 1567, 33%] with alpha = 1.3.</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="co">#&gt; ● Beta C1 [n = 1636, 32%] with mean = 0.85.</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; ● Beta C2 [n = 1797, 35%] with mean = 0.55.</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; ℹ Score(s): NLL = -2558.66; ICL = -4657.61 (-5009.64), H = 383.05 (31.02). Fit converged by MM in 28</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; steps.</span></a></code></pre></div>
<p>A call of <code>mobster_fit</code> will return a list with 3 elements:</p>
<ul>
<li>the best fit <code>fit$best</code>, according to the selected scoring method;</li>
<li>
<code>fit$runs</code>, a list with the ranked fits; <code>best</code> matches the head of this list;</li>
<li>
<code>fit$fits.table</code>, a table that summarises the scores for each one of the runs.</li>
</ul>
<p>Each fit object (<code>best</code> or any object stored in <code>runs</code>) is from the S3 class <code>dbpmm</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="co"># Print the best model</span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(fit<span class="op">$</span>best)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 2 Beta(s) and a tail ─────────────────────────────────</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="co">#&gt; ● Clusters: π = 35% [C2], 33% [Tail] and 32% [C1], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="co">#&gt; ● Tail [n = 1567, 33%] with alpha = 1.3.</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="co">#&gt; ● Beta C1 [n = 1636, 32%] with mean = 0.85.</span></a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co">#&gt; ● Beta C2 [n = 1797, 35%] with mean = 0.55.</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co">#&gt; ℹ Score(s): NLL = -2558.66; ICL = -4657.61 (-5009.64), H = 383.05 (31.02). Fit converged by MM in 28</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="co">#&gt; steps.</span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="co"># Print top-3 models</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(fit<span class="op">$</span>runs[[<span class="dv">1</span>]]) </a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 2 Beta(s) and a tail ─────────────────────────────────</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="co">#&gt; ● Clusters: π = 35% [C2], 33% [Tail] and 32% [C1], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co">#&gt; ● Tail [n = 1567, 33%] with alpha = 1.3.</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="co">#&gt; ● Beta C1 [n = 1636, 32%] with mean = 0.85.</span></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co">#&gt; ● Beta C2 [n = 1797, 35%] with mean = 0.55.</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18"><span class="co">#&gt; ℹ Score(s): NLL = -2558.66; ICL = -4657.61 (-5009.64), H = 383.05 (31.02). Fit converged by MM in 28</span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="co">#&gt; steps.</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(fit<span class="op">$</span>runs[[<span class="dv">2</span>]])</a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 1 Beta(s) and a tail ─────────────────────────────────</span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="co">#&gt; ● Clusters: π = 71% [C1] and 29% [Tail], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="co">#&gt; ● Tail [n = 1429, 29%] with alpha = 1.5.</span></a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="co">#&gt; ● Beta C1 [n = 3571, 71%] with mean = 0.68.</span></a>
<a class="sourceLine" id="cb7-25" data-line-number="25"><span class="co">#&gt; ℹ Score(s): NLL = -1446.64; ICL = -2379.16 (-2842.18), H = 463.01 (0). Fit converged by MM in 22 steps.</span></a>
<a class="sourceLine" id="cb7-26" data-line-number="26"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(fit<span class="op">$</span>runs[[<span class="dv">3</span>]])</a>
<a class="sourceLine" id="cb7-27" data-line-number="27"><span class="co">#&gt; ── [ MOBSTER ] My MOBSTER model n = 5000 with k = 1 Beta(s) and a tail ─────────────────────────────────</span></a>
<a class="sourceLine" id="cb7-28" data-line-number="28"><span class="co">#&gt; ● Clusters: π = 71% [C1] and 29% [Tail], with π &gt; 0.</span></a>
<a class="sourceLine" id="cb7-29" data-line-number="29"><span class="co">#&gt; ● Tail [n = 1429, 29%] with alpha = 1.5.</span></a>
<a class="sourceLine" id="cb7-30" data-line-number="30"><span class="co">#&gt; ● Beta C1 [n = 3571, 71%] with mean = 0.68.</span></a>
<a class="sourceLine" id="cb7-31" data-line-number="31"><span class="co">#&gt; ℹ Score(s): NLL = -1446.64; ICL = -2379.17 (-2842.18), H = 463.01 (0). Fit converged by MM in 22 steps.</span></a></code></pre></div>
<p>Usually, one keeps working with the <code>best</code> model fit. From that it is possible to extract the results of the fit, and the clustering assignments. The output is a copy of the input data, with a column reporting the model’s latent variables (LVs) and the <code>cluster</code> assignment (<em>hard clustering</em>).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># All assignments</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="kw"><a href="../reference/Clusters.html">Clusters</a></span>(fit<span class="op">$</span>best)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">#&gt; # A tibble: 5,000 x 5</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="co">#&gt;      VAF cluster    Tail    C1           C2</span></a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt;    &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt;  1 0.856 C1      0.00371 0.996 0.000000547 </span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt;  2 0.853 C1      0.00375 0.996 0.000000759 </span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="co">#&gt;  3 0.844 C1      0.00403 0.996 0.00000242  </span></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co">#&gt;  4 0.827 C1      0.00520 0.995 0.0000187   </span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co">#&gt;  5 0.855 C1      0.00371 0.996 0.000000571 </span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11"><span class="co">#&gt;  6 0.854 C1      0.00373 0.996 0.000000684 </span></a>
<a class="sourceLine" id="cb8-12" data-line-number="12"><span class="co">#&gt;  7 0.878 C1      0.00415 0.996 0.0000000246</span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"><span class="co">#&gt;  8 0.865 C1      0.00370 0.996 0.000000171 </span></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co">#&gt;  9 0.845 C1      0.00398 0.996 0.00000208  </span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15"><span class="co">#&gt; 10 0.874 C1      0.00392 0.996 0.0000000480</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16"><span class="co">#&gt; # … with 4,990 more rows</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17"></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="co"># Assignments with LVs probability above 85%</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19"><span class="kw"><a href="../reference/Clusters.html">Clusters</a></span>(fit<span class="op">$</span>best, <span class="dt">cutoff_assignment =</span> <span class="fl">0.85</span>)</a>
<a class="sourceLine" id="cb8-20" data-line-number="20"><span class="co">#&gt; # A tibble: 5,000 x 5</span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="co">#&gt;      VAF cluster    Tail    C1           C2</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="co">#&gt;    &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb8-23" data-line-number="23"><span class="co">#&gt;  1 0.856 C1      0.00371 0.996 0.000000547 </span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24"><span class="co">#&gt;  2 0.853 C1      0.00375 0.996 0.000000759 </span></a>
<a class="sourceLine" id="cb8-25" data-line-number="25"><span class="co">#&gt;  3 0.844 C1      0.00403 0.996 0.00000242  </span></a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="co">#&gt;  4 0.827 C1      0.00520 0.995 0.0000187   </span></a>
<a class="sourceLine" id="cb8-27" data-line-number="27"><span class="co">#&gt;  5 0.855 C1      0.00371 0.996 0.000000571 </span></a>
<a class="sourceLine" id="cb8-28" data-line-number="28"><span class="co">#&gt;  6 0.854 C1      0.00373 0.996 0.000000684 </span></a>
<a class="sourceLine" id="cb8-29" data-line-number="29"><span class="co">#&gt;  7 0.878 C1      0.00415 0.996 0.0000000246</span></a>
<a class="sourceLine" id="cb8-30" data-line-number="30"><span class="co">#&gt;  8 0.865 C1      0.00370 0.996 0.000000171 </span></a>
<a class="sourceLine" id="cb8-31" data-line-number="31"><span class="co">#&gt;  9 0.845 C1      0.00398 0.996 0.00000208  </span></a>
<a class="sourceLine" id="cb8-32" data-line-number="32"><span class="co">#&gt; 10 0.874 C1      0.00392 0.996 0.0000000480</span></a>
<a class="sourceLine" id="cb8-33" data-line-number="33"><span class="co">#&gt; # … with 4,990 more rows</span></a></code></pre></div>
<p>The second call imposes a cut to the assignments with less than 85% probability mass in the LVs.</p>
<p>If you want to assign some new data to the fit model you can use function <code>Clusters_denovo</code>.</p>
</div>
<div id="basic-plots-of-a-fit" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-plots-of-a-fit" class="anchor"></a>Basic plots of a fit</h1>
<p>Clusters can be plot as an histogram with the model density (total and per mixture). By default, <code>mobster</code> names Beta clusters <code>C1</code>, <code>C2</code>, etc. according to the decreasing order of their mean; so <code>C1</code> is always the cluster with highest Beta mean, etc. If the data are diploid mutations, <code>C1</code> should represent clonal mutations.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># Plot the best model</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(fit<span class="op">$</span>best)</a></code></pre></div>
<p><img src="a1_introduction_files/figure-html/unnamed-chunk-10-1.png" width="480"></p>
<p>A comparative plot between the fit and data is assembled using <a href="https://cran.r-project.org/web/packages/cowplot/vignettes/introduction.html">cowplot</a>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">cowplot<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">  dataset<span class="op">$</span>plot, </a>
<a class="sourceLine" id="cb10-3" data-line-number="3">  <span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(fit<span class="op">$</span>best), </a>
<a class="sourceLine" id="cb10-4" data-line-number="4">  <span class="dt">ncol =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">  <span class="dt">align =</span> <span class="st">'h'</span>)</a></code></pre></div>
<p><img src="a1_introduction_files/figure-html/unnamed-chunk-11-1.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#input-data-for-mobster">Input data for <code>mobster</code></a><ul class="nav nav-pills nav-stacked">
<li><a href="#driver-annotations">Driver annotations</a></li>
      </ul>
</li>
      <li><a href="#generating-random-models-and-data">Generating random models and data</a></li>
      <li><a href="#fitting-a-dataset">Fitting a dataset</a></li>
      <li><a href="#basic-plots-of-a-fit">Basic plots of a fit</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
