<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>7. Visualizing subclonal expansions • mobster</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="7. Visualizing subclonal expansions">
<meta property="og:description" content="mobster">
<meta property="og:image" content="caravagn.github.io/mobster/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mobster</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/mobster.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1_introduction.html">1. Introduction</a>
    </li>
    <li>
      <a href="../articles/a2_plotting.html">2. Plotting fits</a>
    </li>
    <li>
      <a href="../articles/a3_bootstrap.html">3. Confidence estimation by the bootstrap</a>
    </li>
    <li>
      <a href="../articles/a4_popgen.html">4. Population Genetics statistics</a>
    </li>
    <li>
      <a href="../articles/a5_dnds.html">5. dN/dS statistics</a>
    </li>
    <li>
      <a href="../articles/a6_CloneTrees.html">6. Clone trees from clusters</a>
    </li>
    <li>
      <a href="../articles/a7_Example_tumour_simulation.html">7. Visualizing subclonal expansions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/mobster/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="a7_Example_tumour_simulation_files/header-attrs-2.1/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>7. Visualizing subclonal expansions</h1>
                        <h4 class="author">Giulio Caravagna and Marc Williams</h4>
            
            <h4 class="date">10 mai, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/mobstervignettes/a7_Example_tumour_simulation.Rmd"><code>vignettes/a7_Example_tumour_simulation.Rmd</code></a></small>
      <div class="hidden name"><code>a7_Example_tumour_simulation.Rmd</code></div>

    </div>

    
    
<center>
<a href="https://t-heide.github.io/TEMULATOR/"><img src="https://t-heide.github.io/TEMULATOR/reference/figures/logo.png" width="77px" height="91px"></a>
</center>
<p><br></p>
<p>Stochastic branching processes models of tumour growth describe cell division processes and the propagation of somatic alleles that accrue during tumour evolution.</p>
<p>In this RMarkdown, we use these population genetics models to visualise the joint effects of <em>neutral evolution</em> and <em>positive selection</em> on sequencing data of tumour’s Variant Allele Frequencies (VAFs).</p>
<p>You can generate similar dynamics using the R package <a href="https://t-heide.github.io/TEMULATOR">Tumour Emulator (TEMULATOR)</a> or the Julia package <a href="https://github.com/marcjwilliams1/CancerSeqSim">CancerSeqSim</a>.</p>
<div id="simulated-tumour-growth" class="section level1">
<h1 class="hasAnchor">
<a href="#simulated-tumour-growth" class="anchor"></a>Simulated tumour growth</h1>
<p>We have simulated a tumour with 2 clones. The tumours starts with the ancestral clone at time <code>t = 0</code>, and the subclone with increased positive selection relative to its ancestor (<span class="math inline">\(s&gt;0\)</span>), emerging at time <code>t = 9</code>.</p>
<p>We load the simulation object <a href="./simulationdata.Rdata">simulationdata.Rdata</a>.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">mobster</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">dplyr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">tibble</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">require</a></span>(<span class="no">ggplot2</span>)

<span class="co"># Load input data from Marc</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">'simulationdata.Rdata'</span>)

<span class="co"># Split by time point (`d` variable)</span>
<span class="no">df.steps</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span>(<span class="no">df</span>, <span class="kw">f</span> <span class="kw">=</span> <span class="no">df</span>$<span class="no">d</span>)

<span class="co"># Integer times</span>
<span class="no">t_integers</span> <span class="kw">=</span> <span class="no">df</span>$<span class="no">d</span> <span class="kw">%&gt;%</span>  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span>() <span class="kw">%&gt;%</span> <span class="no">unique</span> <span class="kw">%&gt;%</span> <span class="no">sort</span>

<span class="no">df</span> <span class="kw">=</span> <span class="no">df</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>() <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">d</span> <span class="kw">%in%</span> <span class="no">t_integers</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">cloneid</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">cloneid</span>))

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">df</span> <span class="kw">%&gt;%</span> <span class="no">as_tibble</span>)
<span class="co">#&gt; # A tibble: 70,474 x 6</span>
<span class="co">#&gt;      VAF trueVAF cloneid cloneVAF     d popsize</span>
<span class="co">#&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 0.496     0.5 1              0     1       1</span>
<span class="co">#&gt;  2 0.459     0.5 1              0     1       1</span>
<span class="co">#&gt;  3 0.496     0.5 1              0     1       1</span>
<span class="co">#&gt;  4 0.5       0.5 1              0     1       1</span>
<span class="co">#&gt;  5 0.463     0.5 1              0     1       1</span>
<span class="co">#&gt;  6 0.460     0.5 1              0     1       1</span>
<span class="co">#&gt;  7 0.484     0.5 1              0     1       1</span>
<span class="co">#&gt;  8 0.474     0.5 1              0     1       1</span>
<span class="co">#&gt;  9 0.491     0.5 1              0     1       1</span>
<span class="co">#&gt; 10 0.505     0.5 1              0     1       1</span>
<span class="co">#&gt; # ... with 70,464 more rows</span></pre></body></html></div>
<p>We first use <code>gganimate</code> to create a GIF of the simulated data.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="no">ani</span> <span class="kw">=</span> <span class="no">df</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reexports.html">as_tibble</a></span>()  <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="kw">cloneid</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="no">cloneid</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">VAF</span>, <span class="kw">fill</span> <span class="kw">=</span> <span class="no">cloneid</span>)) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_histogram.html">geom_histogram</a></span>(<span class="kw">binwidth</span> <span class="kw">=</span> <span class="fl">0.01</span>) +
  <span class="kw pkg">gganimate</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gganimate/man/transition_states.html">transition_states</a></span>(
    <span class="no">d</span>,
    <span class="kw">transition_length</span> <span class="kw">=</span> <span class="fl">2</span>,
    <span class="kw">state_length</span> <span class="kw">=</span> <span class="fl">2</span>
    ) +
  <span class="kw pkg">gganimate</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gganimate/man/ease_aes.html">ease_aes</a></span>(<span class="st">'cubic-in-out'</span>) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">ylab</a></span>(<span class="st">"Counts (number of mutations)"</span>) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span>(
    <span class="kw">title</span> <span class="kw">=</span> <span class="st">'Simulated VAF distribution'</span>,
    <span class="kw">subtitle</span> <span class="kw">=</span> <span class="st">'Tumour doublings (clock): t = {closest_state}'</span>
    ) +
  <span class="kw pkg">ggthemes</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggthemes/man/scale_ptol.html">scale_fill_ptol</a></span>() +
  <span class="kw pkg">mobster</span><span class="kw ns">:::</span><span class="fu">my_ggplot_theme</span>()

  <span class="kw pkg">gganimate</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gganimate/man/animate.html">animate</a></span>(<span class="no">ani</span>,
                   <span class="kw">width</span> <span class="kw">=</span> <span class="fl">480</span>,
                   <span class="kw">height</span> <span class="kw">=</span> <span class="fl">480</span>,
                   <span class="kw">renderer</span> <span class="kw">=</span> <span class="kw pkg">gganimate</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/gganimate/man/renderers.html">magick_renderer</a></span>())</pre></body></html></div>
<p><img src="a7_Example_tumour_simulation_files/figure-html/unnamed-chunk-3-1.gif"><!-- --></p>
<p>The time units of the plot are <em>genome doublings</em>. We have simulate Poisson-distributed <em>whole-genome sequencing</em> (WGS) data at <code>120x</code> median depth, and perfect tumour purity (<code>100%</code>). These parameters affect the observable dynamics; here the subclone is visible after <code>t = 14</code> doublings, and sweeps completely at <code>t = 19</code>.</p>
<p>We can plot the data distribution at each of a discrete set of time-points.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">df</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">VAF</span>, <span class="kw">fill</span> <span class="kw">=</span> <span class="no">cloneid</span>)) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_histogram.html">geom_histogram</a></span>(<span class="kw">binwidth</span> <span class="kw">=</span> <span class="fl">0.01</span>) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/ggtheme.html">theme_light</a></span>() +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/facet_wrap.html">facet_wrap</a></span>( ~ <span class="no">d</span>, <span class="kw">nrow</span> <span class="kw">=</span> <span class="fl">3</span>) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/geom_abline.html">geom_vline</a></span>(<span class="kw">xintercept</span> <span class="kw">=</span> <span class="fl">0.05</span>,
             <span class="kw">linetype</span> <span class="kw">=</span> <span class="st">'dashed'</span>,
             <span class="kw">size</span> <span class="kw">=</span> <span class="fl">.3</span>) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span>(
    <span class="kw">title</span> <span class="kw">=</span> <span class="st">'Snapshots of tumour growth with positive subclonal selection dynamics'</span>,
    <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Counts"</span>,
    <span class="kw">x</span> <span class="kw">=</span> <span class="st">'Simulated VAF'</span>
    ) +
  <span class="kw pkg">CNAqc</span><span class="kw ns">:::</span><span class="fu">my_ggplot_theme</span>() +
  <span class="kw pkg">ggthemes</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggthemes/man/scale_ptol.html">scale_fill_ptol</a></span>()</pre></body></html></div>
<p><img src="a7_Example_tumour_simulation_files/figure-html/unnamed-chunk-4-1.png" width="1344"> WGS data are simulated independently at each time-point, and the VAFs at each point are therefore uncorrelated. The monoclonal ancestral population is dark blue; mutations in the new subclone plus the hitchikkers are coloured in purple.</p>
<p>The dynamics of the tumour span from time <code>t=0</code> to <code>t=23</code>; the subclone is undetectable before <code>t=14</code>, and has sweeped through at the end of the simulation.</p>
</div>
<div id="mobster-fits" class="section level1">
<h1 class="hasAnchor">
<a href="#mobster-fits" class="anchor"></a>MOBSTER fits</h1>
<p>We capture the dynamics (fit and plot) with a snapshot function <code>snap</code>.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">snap</span> <span class="kw">=</span> <span class="kw">function</span>(<span class="no">data</span>, <span class="no">t</span>, <span class="no">...</span>)
{
  <span class="no">x</span> <span class="kw">=</span> <span class="no">data</span><span class="kw">[[</span><span class="no">t</span>]]
  <span class="no">x</span> <span class="kw">=</span> <span class="no">x</span>[<span class="no">x</span>$<span class="no">VAF</span> <span class="kw">&gt;</span> <span class="fl">0.05</span>,]

  <span class="no">time_label</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">'t_0 = '</span>, <span class="no">x</span>$<span class="no">d</span>[<span class="fl">1</span>])

  <span class="co"># Ellipsis to get what fits we want</span>
  <span class="no">f</span> <span class="kw">=</span> <span class="fu"><a href="../reference/mobster_fit.html">mobster_fit</a></span>(<span class="no">x</span>, <span class="kw">description</span> <span class="kw">=</span> <span class="no">time_label</span>, <span class="no">...</span>)$<span class="no">best</span>

  <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">fit</span> <span class="kw">=</span> <span class="no">f</span>, <span class="kw">plot</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">f</span>) + <span class="fu"><a href="https://rdrr.io/pkg/ggplot2/man/labs.html">labs</a></span>(<span class="kw">title</span> <span class="kw">=</span> <span class="no">time_label</span>)))
}

<span class="co"># Initial clone (monoclonal)</span>
<span class="no">initiation</span> <span class="kw">=</span> <span class="fu">snap</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">df.steps</span>, <span class="kw">t</span> <span class="kw">=</span> <span class="fl">5</span>, <span class="kw">K</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">samples</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt;  [ MOBSTER fit ]</span>
<span class="co">#&gt; v Loaded input data, n = 2151.</span>
<span class="co">#&gt; &gt; n = 2151. Mixture with k = 1 Beta(s). Pareto tail: TRUE and FALSE. Output</span>
<span class="co">#&gt; clusters with p &gt; 0.02 and n &gt; 10.</span>
<span class="co">#&gt; &gt; Custom fit by Moments-matching in up to 250 steps, with e = 1e-10 and peaks</span>
<span class="co">#&gt; initialisation.</span>
<span class="co">#&gt; &gt; Scoring (without parallel) 1 x 1 x 2 = 2 models by reICL.</span>
<span class="co">#&gt; i MOBSTER fits completed in 703ms.</span>
<span class="co">#&gt; -- [ MOBSTER ] t_0 = 3 n = 2151 with k = 1 Beta(s) and a tail ------------------</span>
<span class="co">#&gt; * Clusters: p = 56% [Tail] and 44% [C1], with p &gt; 0.</span>
<span class="co">#&gt; * Tail [n = 1156, 56%] with alpha = 1.2.</span>
<span class="co">#&gt; * Beta C1 [n = 995, 44%] with mean = 0.5.</span>
<span class="co">#&gt; i Score(s): NLL = -1885.56; ICL = -3543.5 (-3725.09), H = 181.58 (0). Fit</span>
<span class="co">#&gt; converged by MM in 21 steps.</span>

<span class="co"># Ongoing subclonal expansion (polyclonal)</span>
<span class="no">selection</span> <span class="kw">=</span>  <span class="fu">snap</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">df.steps</span>, <span class="kw">t</span> <span class="kw">=</span> <span class="fl">34</span>, <span class="kw">K</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">samples</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt;  [ MOBSTER fit ]</span>
<span class="co">#&gt; v Loaded input data, n = 3449.</span>
<span class="co">#&gt; &gt; n = 3449. Mixture with k = 2 Beta(s). Pareto tail: TRUE and FALSE. Output</span>
<span class="co">#&gt; clusters with p &gt; 0.02 and n &gt; 10.</span>
<span class="co">#&gt; &gt; Custom fit by Moments-matching in up to 250 steps, with e = 1e-10 and peaks</span>
<span class="co">#&gt; initialisation.</span>
<span class="co">#&gt; &gt; Scoring (without parallel) 1 x 1 x 2 = 2 models by reICL.</span>
<span class="co">#&gt; i MOBSTER fits completed in 14.2s.</span>
<span class="co">#&gt; -- [ MOBSTER ] t_0 = 17.5 n = 3449 with k = 2 Beta(s) and a tail ---------------</span>
<span class="co">#&gt; * Clusters: p = 49% [C2], 26% [Tail], and 25% [C1], with p &gt; 0.</span>
<span class="co">#&gt; * Tail [n = 805, 26%] with alpha = 1.2.</span>
<span class="co">#&gt; * Beta C1 [n = 967, 25%] with mean = 0.51.</span>
<span class="co">#&gt; * Beta C2 [n = 1677, 49%] with mean = 0.33.</span>
<span class="co">#&gt; i Score(s): NLL = -2141.73; ICL = -3177.64 (-3783.57), H = 1032.51 (426.58).</span>
<span class="co">#&gt; Fit interrupted by MM in 250 steps.</span>

<span class="co"># Sweeped clone (monoclonal)</span>
<span class="no">sweep</span> <span class="kw">=</span>  <span class="fu">snap</span>(<span class="kw">data</span> <span class="kw">=</span> <span class="no">df.steps</span>, <span class="kw">t</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">df.steps</span>), <span class="kw">K</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">samples</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="co">#&gt;  [ MOBSTER fit ]</span>
<span class="co">#&gt; v Loaded input data, n = 4001.</span>
<span class="co">#&gt; &gt; n = 4001. Mixture with k = 1 Beta(s). Pareto tail: TRUE and FALSE. Output</span>
<span class="co">#&gt; clusters with p &gt; 0.02 and n &gt; 10.</span>
<span class="co">#&gt; &gt; Custom fit by Moments-matching in up to 250 steps, with e = 1e-10 and peaks</span>
<span class="co">#&gt; initialisation.</span>
<span class="co">#&gt; &gt; Scoring (without parallel) 1 x 1 x 2 = 2 models by reICL.</span>
<span class="co">#&gt; i MOBSTER fits completed in 1s.</span>
<span class="co">#&gt; -- [ MOBSTER ] t_0 = 23 n = 4001 with k = 1 Beta(s) and a tail -----------------</span>
<span class="co">#&gt; * Clusters: p = 60% [C1] and 40% [Tail], with p &gt; 0.</span>
<span class="co">#&gt; * Tail [n = 1530, 40%] with alpha = 1.1.</span>
<span class="co">#&gt; * Beta C1 [n = 2471, 60%] with mean = 0.49.</span>
<span class="co">#&gt; i Score(s): NLL = -3417.33; ICL = -6427.2 (-6784.89), H = 357.69 (0). Fit</span>
<span class="co">#&gt; converged by MM in 28 steps.</span>

<span class="kw pkg">cowplot</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(<span class="no">initiation</span>$<span class="no">plot</span>, <span class="no">selection</span>$<span class="no">plot</span>, <span class="no">sweep</span>$<span class="no">plot</span>, <span class="kw">nrow</span> <span class="kw">=</span> <span class="fl">1</span>)</pre></body></html></div>
<p><img src="a7_Example_tumour_simulation_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
<p>Notice that the tail at the end of the simulation is now mostly dominated by the progeny of the most recent clone, and the amount of mutations in the tail and the clonal cluster is balanced as in the beginning of the simulation. This because in this simulation the mutation rate of the tumour is the same across all simulated clones.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
